src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/DagController.cs

```diff
@@ -0,0 +1,133 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using MultiFormats;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Catalyst.Core.Modules.Dfs.WebApi.V0.Controllers
+{
+    /// <summary>
+    ///   A link to a CID.
+    /// </summary>
+    public class LinkedDataDto
+    {
+        /// <summary>
+        ///   The CID.
+        /// </summary>
+        [JsonProperty(PropertyName = "/")]
+        public string Link;
+    }
+
+    /// <summary>
+    ///   A CID as linked data.
+    /// </summary>
+    public class LinkedDataCidDto
+    {
+        /// <summary>
+        ///   A link to the CID.
+        /// </summary>
+        public LinkedDataDto Cid;
+    }
+
+    /// <summary>
+    ///   Manages the IPLD (linked data) Directed Acrylic Graph.
+    /// </summary>
+    public class DagController : IpfsController
+    {
+        /// <summary>
+        ///   Creates a new controller.
+        /// </summary>
+        public DagController(IDfsService dfs) : base(dfs) { }
+
+        /// <summary>
+        ///  Resolve a reference (NYI).
+        /// </summary>
+        [HttpGet, HttpPost, Route("dag/resolve")] // TODO
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)